### PR TITLE
Add UTC support for failing DayOne tests

### PR DESCRIPTION
**Pull request by MinchinWeb**
Travis-CI (typically) runs in UTC, and UTC behaves weird compared to other timezones. This adds UTC support to the DayOne parts of the code.

Thus this re-enables the (most of) DayOne tests (all of which broke previously, because of UTC), building on #742 .

The code change is pulled from #350.

### Checklist
- [x] The code change is tested and works locally.
- [x] Tests pass. Your PR cannot be merged unless tests pass
- [x] There is no commented out code in this PR.
- [x] Have you followed the guidelines in our Contributing document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](../pulls) for the same update/change?
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [x] Have you written new tests for your core changes, as applicable?

### CODE REVIEW

File: jrnl/DayOneJournal.py [hunk 1]
```diff
@@ -52,7 +52,16 @@ def open(self):
                     except (KeyError, pytz.exceptions.UnknownTimeZoneError):
                         timezone = tzlocal.get_localzone()
                     date = dict_entry["Creation Date"]
-                    date = date + timezone.utcoffset(date, is_dst=False)
+                    try:
```
Comment (wren) [inline jrnl/DayOneJournal.py]: Passing on errors is an anti-pattern I'd like to get us out of whenever possible. 

How about something like this?
```
if dict_entry.get("Time Zone") in pytz.all_timezones:
    timezone = pytz.timezone(dict_entry["Time Zone"])
else:
    timezone = tzlocal.get_localzone()

date = dict_entry["Creation Date"]
if timezone.zone != "UTC":
    date = date + timezone.utcoffset(date, is_dst=False)
```

Comment (wren) [reply jrnl/DayOneJournal.py]: Sorry, apparently didn't hit submit to the above comment yesterday (found it sitting in a tab today).

Comment (MinchinWeb) [reply jrnl/DayOneJournal.py]: I agree that generally passing on an a raised error is bad from, but what about cases like this where the raised error is actually telling us things are working as expected? This from also supports the philosophy of "it's easier to ask for forgiveness than permission", which is often typically idiomatic Python.

The [article](https://devblogs.microsoft.com/python/idiomatic-python-eafp-versus-lbyl/) I found on this (first link on Google) gives an example and explicitly uses the `except: pass` formation. The [Python docs](https://docs.python.org/3.8/glossary.html#term-eafp) seem to support this form as well.

So I think we should leave it as it is as being more idiomatic Python.

Comment (MinchinWeb) [reply jrnl/DayOneJournal.py]: @wren : a decision here is what's holding to #786. I like it this way, and think it's more "Pythonic", but of you still want it changed, I'm happy to make the change. Just key me know.

Comment (wren) [reply jrnl/DayOneJournal.py]: Yes, please. Thank you!

**Comment by MinchinWeb** [comment]
@wren: changed as requested! This, and #786 should be good to go!

**Pull request merged**